packages/compiler/config/config-schema.ts

```diff
@@ -9,6 +18,31 @@ export const CadlConfigJsonSchema: JSONSchemaType<CadlRawConfig> = {
       type: "string",
       nullable: true,
     },
+    "environment-variables": {
+      type: "object",
+      nullable: true,
+      required: [],
+      additionalProperties: {
+        type: "object",
+        properties: {
+          default: { type: "string" },
+        },
+        required: ["default"],
+      },
+    },
+    parameters: {
+      type: "object",
+      nullable: true,
+      required: [],
+      additionalProperties: {
+        type: "object",
+        properties: {
+          default: { type: "string" },
```

Comment:
Idea is that, it would be anoyoing to require an arg to always be passed in order to build. That also means that this might prevent ide support in some cases.

Reply:
I personally would prefer no environment variables at all, but I guess I can settle for forced default.

Please don't make it easy to create Cadl projects that only work when loaded in the right environment variables set and break the IDE. This is exactly what happened in various Microsoft build systems and it was a disaster for working in Visual Studio.